flycheck.el

```diff
@@ -2930,31 +2956,144 @@ Slots:
      in order to be able to present them to the user.
 
      See `flycheck-related-errors`."
-  buffer checker filename line column message level id group)
+  buffer checker filename coords region message level id group)
+
+; Compute and cache coords if nil
+(advice-add 'flycheck-error-coords :after-until
+            (lambda (err)
+              (let ((coords (flycheck-region-to-coords (flycheck-error-region err)
+                                                       (flycheck-error-buffer err))))
+                (setf (flycheck-error-coords err) coords)
+                coords)))
```

Comment:
Advice is typically reserved for user code, or desperate situations; let's find a different way to do this.